### PR TITLE
media-video/ffmpeg: add 6.1; wire up vulkan filters

### PR DESCRIPTION
**Pull request by sevz17**

### CODE REVIEW

**Comment by sevz17** [comment]
@thesamesam

**Comment by thesamesam** [comment]
Not checked yet but have you made sure to both use 9999 as a basis for 6.1, but also diff any configure/makefile changes between 6.x and 6.1.x when preparing 6.1?

**Comment by sevz17** [comment]
I used 9999 for 6.1, but not checked for configure/makefile changes, since I use 9999 and it does not seem to have problems I didn't think it was necessary

Checking right now.

**Comment by thesamesam** [comment]
Definitely necessary for either automagic stuff, new min. version requirements, or features which are off-by-default we want to wire up

**Comment by sevz17** [comment]
```diff
+  --enable-libaribcaption  enable ARIB text and caption decoding via libaribcaption [no]
```
libaribcaption is not packaged (yet?)


```diff
+  --disable-dotprod        disable DOTPROD optimizations
+  --disable-i8mm           disable I8MM optimizations
```
seems like they are arm cpu instructions but there are no CPU_FLAGS_*  

also seems like it requires spirv_compiler for some new vulkan filters (and also for old filters) but tbh I don't know to fix it  

harfbuzz is a new dependency for the drawtext filter (which already requires
freetype), I was thinking in rename USE=truetype to USE=drawtext and the required deps but freetype is also (optionally) used in another filter, what is be the correct way of doing that?  

As for new min versions seems like they didn't change since https://github.com/gentoo/gentoo/pull/33322

**Comment by thesamesam** [comment]
> ```diff
> +  --enable-libaribcaption  enable ARIB text and caption decoding via libaribcaption [no]
> ```
> 
> libaribcaption is not packaged (yet?)

OK, disable the option with a comment saying that.

> 
> ```diff
> +  --disable-dotprod        disable DOTPROD optimizations
> +  --disable-i8mm           disable I8MM optimizations
> ```
> 
> seems like they are arm cpu instructions but there are no CPU_FLAGS_*

File a bug for both to have a new CPU_FLAGS_ARM USE_EXPAND and then disable in the ebuild with a comment referencing the new bugs (file a new one for both).

> 
> also seems like it requires spirv_compiler for some new vulkan filters (and also for old filters) but tbh I don't know to fix it
> 

I'm not sure what you mean by this bit. What do you see?

> harfbuzz is a new dependency for the drawtext filter (which already requires freetype), I was thinking in rename USE=truetype to USE=drawtext and the required deps but freetype is also (optionally) used in another filter, what is be the correct way of doing that?

Is someone really likely to want the filters separately? Most people will have ft/hf anyway and it's kind of hard to have one without the other too. Maybe just keep it as USE=truetype and update the description in metadata.xml accordingly?

> 
> As for new min versions seems like they didn't change since #33322

OK.

I may cherry-pick the non-6.1 bits for now.

**Comment by sevz17** [comment]
> > also seems like it requires spirv_compiler for some new vulkan filters (and also for old filters) but tbh I don't know to fix it
> 
> I'm not sure what you mean by this bit. What do you see?
> 

This
```
nlmeans_vulkan_filter_deps="vulkan spirv_compiler"
```

> > harfbuzz is a new dependency for the drawtext filter (which already requires freetype), I was thinking in rename USE=truetype to USE=drawtext and the required deps but freetype is also (optionally) used in another filter, what is be the correct way of doing that?
> 
> Is someone really likely to want the filters separately?

tbh, I don't know I just installed 9999 because I wanted the vulkan hwdec in mpv :), I know next to nothing about ffmpeg filters.

> Most people will have ft/hf anyway and it's kind of hard to have one without the other too. Maybe just keep it as USE=truetype and update the description in metadata.xml accordingly?
> 

Done, after thinking about it more, renaming the USE would probably affect ebuilds which depend on ffmpeg... too much work.

File: media-video/ffmpeg/ffmpeg-9999.ebuild [hunk 1]
```diff
@@ -468,11 +473,16 @@ multilib_src_configure() {
 	done
 
 	# LTO support, bug #566282, bug #754654, bug #772854
-	[[ ${ABI} != x86 ]] && is-flagq "-flto*" && myconf+=( "--enable-lto" )
+	if [[ ${ABI} != x86 ]] && is-flagq "-flto*"; then
+		# Respect -flto value, e.g -flto=thin
+		local v="$(get-flag flto)"
+		[[ -n ${v} ]] && myconf+=( "--enable-lto=${v}" ) || myconf+=( "--enable-lto" )
```
Comment (sevz17) [inline media-video/ffmpeg/ffmpeg-9999.ebuild:479]: python could also benefit of this: https://github.com/python/cpython/issues/88506

**Comment by sevz17** [comment]
Nevermind, spirv_compiler requires either shaderc or glslang. Updating ebuild (and checking if it also happens to older releases)

**Pull request closed without merging**